core/DataTable/Map.php

```diff
@@ -540,4 +540,25 @@ public function getColumns()
         }
         return array();
     }
+
+
+    /**
+     * Return the metadata of the latest item in the datable array.
+     * @return array
+     * @throws Exception
+     */
+    public function getLatestMetaData()
+    {
+
+        $largestKey = null;
+        // find the largest key
+        foreach ($this->array as $key => $item) {
+            if (!$largestKey || (strtotime($largestKey) < strtotime($key))) {
+                $largestKey = $key;
+            }
+        }
+
+        $latestItem = $this->array[$largestKey];
```

Comment:
For an empty map this method would throw a warning or error, as it would try to access `$this->array[null];`